jbpm-test-coverage/src/test/java/org/jbpm/test/functional/task/ProcessHumanTaskTest.java

```diff
@@ -83,6 +91,39 @@ public void testProcess() {
 		assertProcessInstanceNotActive(processInstance.getId(), ksession);
 	}
 	
+    @Test
+    public void testProcessWithHumanTaskAndCustomStrategy() {
+        createRuntimeManager("org/jbpm/test/functional/task/HumanTaskWithCustomAssignmentStrategy.bpmn2");
+        RuntimeEngine runtimeEngine = getRuntimeEngine();
+        KieSession ksession = runtimeEngine.getKieSession();
+        TaskService taskService = runtimeEngine.getTaskService();
+        AssignmentServiceRegistry.get().addStrategy(new AssignmentStrategy() {
+
+            @Override
+            public String getIdentifier() {
+                return "CustomStrategy";
+            }
+
+            @Override
+            public Assignment apply(Task task, TaskContext context, String excludedUser) {
+                return new AssignmentImpl("doctor");
+            }
+            
+        });
+        AssignmentServiceProvider.get().setEnabled(true);
+        ProcessInstance processInstance = ksession.startProcess("humanTaskWithCustomStrategy", Collections.singletonMap("processHTInput", "CustomStrategy"));
+
+        List<Long> listIds = taskService.getTasksByProcessInstanceId(processInstance.getId());
+        List<Task> list = listIds.stream().map(e -> taskService.getTaskById(e)).collect(Collectors.toList());
+        Task task = list.get(0);
+        // john is potential owner but our custom strategy is overriding it.
+        logger.info("doctor is executing task {}", task.getName());
+        taskService.start(task.getId(), "doctor");
```

Comment:
I would maybe add a call to `taskService.getTasksAssignedAsPotentialOwner` with `doctor` or with `john` to assert the correct task is assigned to the expected person

Reply:
it cannot be started if it is not the owner.